projects/ngx-charts-on-fhir/src/lib/data-layer-list/annotation-options/annotation-options.component.spec.ts

```diff
@@ -19,10 +19,10 @@ const mockColorService = {
   providers: [{ provide: NG_VALUE_ACCESSOR, multi: true, useExisting: forwardRef(() => MockColorPickerComponent) }],
 })
 class MockColorPickerComponent implements ControlValueAccessor {
-  writeValue(obj: any): void { /* TODO document why this method 'writeValue' is empty */ }
-  registerOnChange(fn: any): void { /* TODO document why this method 'registerOnChange' is empty */ }
-  registerOnTouched(fn: any): void { /* TODO document why this method 'registerOnTouched' is empty */ }
-  setDisabledState?(isDisabled: boolean): void { /* TODO document why this method 'setDisabledState' is empty */ }
+  writeValue(obj: any): void { /* Writes a new value to the element. */ }
+  registerOnChange(fn: any): void { /* Registers a callback function that is called when the control's value changes in the UI. */ }
+  registerOnTouched(fn: any): void { /* Registers a callback function that is called by the forms API on initialization to update the form model on blur. */ }
+  setDisabledState?(isDisabled: boolean): void { /* Function that is called by the forms API when the control status changes to or from 'DISABLED'. Depending on the status, it enables or disables the appropriate DOM element */ }
```

Comment:
Sonar flags empty methods because they might be a mistake. The purpose of adding comments to empty methods is to let other developers (and sonar) know that it's not a mistake. Your comments look like they were pulled from the interface documentation and are rather confusing in this context. A simple comment like `/* intentionally empty stub */` would make more sense here.

Reply:
@semicolin will correct it, overthinking to add comments for it.